### PR TITLE
Datadog impl for MDN Redis monitoring

### PR DESCRIPTION
**Pull request by metadave**

### CODE REVIEW

File: apps/mdn/mdn-aws/docs/mdn-dd-redis.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+The secrets titled `datadog-secrets` has 2 key/value pairs: `API_KEY` and `REDIS_CONFIG_FILE`. 
+
+`API_KEY` is a simple base64 encoded value:
+
+```
+echo -n "some_api_key_value" | base64
+```
+
+`REDIS_CONFIG_FILE` is the base64 encoded contents of the [Datadog Redis configuration](https://docs.datadoghq.com/integrations/redisdb/):
```
Comment (jwhitlock) [inline apps/mdn/mdn-aws/docs/mdn-dd-redis.md]: This wasn't what I was expecting. I thought we'd have a ``redisdb.yml.j2`` that was combined with secrets. But, if it works, and is the way preferred by the Datadog image, then 👍

File: apps/mdn/mdn-aws/docs/mdn-dd-redis.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+# To check overall status of the DD agent:
+/etc/init.d/datadog-agent info
+# This command will also show if the Redis agent is running
+```
+
+
+### Datadog Dashboard setup
+
+
+tagged by `mdn_redis_prod_us_west_2`
```
Comment (jwhitlock) [inline apps/mdn/mdn-aws/docs/mdn-dd-redis.md]: This looks like a stray line that should be deleted, or integrated into the next paragraph

File: apps/mdn/mdn-aws/docs/mdn-dd-redis.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
+### How does this work?
```
Comment (jwhitlock) [inline apps/mdn/mdn-aws/docs/mdn-dd-redis.md]: Can we add a "why" paragraph? Something like:

MDN uses Redis as the message transport for Celery, to store tasks that are run in the background. Each Celery queue is stored as a list in Redis. By tracking the length of these lists in Datadog, we can identify trends over time, and alert staff when queued tasks are overwhelming the processing capacity.

File: apps/mdn/mdn-aws/k8s/Makefile [hunk 4]
```diff
@@ -196,6 +196,19 @@ export NEW_RELIC_API_NAME ?= "kuma-backend-${NEW_RELIC_MONITOR_SUFFIX};${NEW_REL
 export NEW_RELIC_KUMASCRIPT_NAME ?= "kumascript-${NEW_RELIC_MONITOR_SUFFIX}"
 export NEW_RELIC_CELERY_NAME ?= "kuma-backend-${NEW_RELIC_MONITOR_SUFFIX};${NEW_RELIC_ROLLUP_NAME}"
 
+export DATADOG_REDIS_DEPLOYMENT_NAME ?= mdn-dd-redis
+export DATADOG_SECRETS_NAME ?= datadog-secrets
+
+export DATADOG_API_KEY ?= foo
+export DATADOG_API_KEY_BASE64 ?= $(shell echo -n "${DATADOG_API_KEY}" | base64)
+
+# cat redisdb.yaml | base64 -w0
```
Comment (jwhitlock) [inline apps/mdn/mdn-aws/k8s/Makefile]: Is this comment redundant?

File: apps/mdn/mdn-aws/docs/mdn-dd-redis.md (hunk 1 above)
Comment (metadave) [reply apps/mdn/mdn-aws/docs/mdn-dd-redis.md]: we have to specify some secret values inside the redisdb.yaml file, and the DD agent Docker image needs the files upon startup to enable the Redis collector. The easiest way to pass in a file with secret data is just by mounting it as a secret. The private MDN repo contains the "rendered" datadog-secrets file, including the contents of redisdb.yaml in base64 format.

File: apps/mdn/mdn-aws/k8s/Makefile [hunk 5]
```diff
@@ -196,6 +196,18 @@ export NEW_RELIC_API_NAME ?= "kuma-backend-${NEW_RELIC_MONITOR_SUFFIX};${NEW_REL
...
+export DATADOG_REDIS_DEPLOYMENT_NAME ?= mdn-dd-redis
+export DATADOG_SECRETS_NAME ?= datadog-secrets
+
+export DATADOG_API_KEY ?= foo
+export DATADOG_API_KEY_BASE64 ?= $(shell echo -n "${DATADOG_API_KEY}" | base64)
+
+export DATADOG_REDIS_CONFIG ?= foo
+# note the base64 -w0, this prevents base64 from splitting the string w/ newlines
+# see https://kubernetes.io/docs/concepts/configuration/secret/
+export DATADOG_REDIS_CONFIG_BASE64 ?= $(shell echo -n "${DATADOG_REDIS_CONFIG}" | base64 -w0)
```
Comment (escattone) [inline apps/mdn/mdn-aws/k8s/Makefile]: @metadave @jwhitlock The `base64` command on my Mac complains about the `-w0` option. It appears that `base64` on Mac is different than on Linux.

Comment (metadave) [reply apps/mdn/mdn-aws/k8s/Makefile]: I've noted that in the doc: https://github.com/mozmeao/infra/pull/561/files#diff-04209630fe5af9be901ec0af15efc1adR28

Comment (metadave) [reply apps/mdn/mdn-aws/k8s/Makefile]: Also, it appears that the osx version of `base64` outputs the correct format by default.

**Pull request merged**